nixos/modules/tasks/network-namespaces.nix

```diff
@@ -0,0 +1,26 @@
+{ config, lib, pkgs, ... }:
+with lib; {
+  options = {
+    networking.namespaces = mkOption {
+      default = [];
+      type = types.listOf types.str;
+      description = ''
+        A list of named network namespaces to create.
+      '';
+    };
+  };
+  config.systemd.services = builtins.listToAttrs (map (netns: {
+    name = "network-namespace-${netns}";
+    value = {
+      description = "Network Namespace: ${netns}";
+      before = [ "network-pre.target" ];
+      wantedBy = [ "multi-user.target" ];
+      serviceConfig = {
+        Type = "oneshot";
+        RemainAfterExit = true;
+        ExecStart = "${pkgs.iproute}/bin/ip netns add ${strings.escapeShellArg netns}";
+        ExecStop = "${pkgs.iproute}/bin/ip netns del ${strings.escapeShellArg netns}";
```

Comment:
According to the `ip-netns` manpage:

```
             It is possible to lose the physical device when it was moved
              to netns and then this netns was deleted with a running
              process:

                 $ ip netns add net0
                 $ ip link set dev eth0 netns net0
                 $ ip netns exec net0 SOME_PROCESS_IN_BACKGROUND
                 $ ip netns del net0

              and eth0 will appear in the default netns only after
              SOME_PROCESS_IN_BACKGROUND will exit or will be killed. To
              prevent this the processes running in net0 should be killed
              before deleting the netns:

                 $ ip netns pids net0 | xargs kill
                 $ ip netns del net0
```

Do you think it's necessary to document this in the module?

Reply:
This is usually used to move `wg*` interfaces into a namespace, right?

So it's probably not too much of an issue. Adding a link to these docs sounds like a good idea, but we don't need to copy this paragraph in.